src/arrayMethodJoin.js

```diff
@@ -5,7 +5,39 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const joinArr = [];
+    let sep = separator;
+    let joinStr = '';
+
+    if (sep === undefined) {
+      sep = ',';
+    }
+
+    if (this.length === 0) {
+      return joinStr;
+    }
+
+    joinArr.length = 2 * this.length - 1;
+
+    for (let i = 0; i < joinArr.length; i++) {
+      if (i % 2 !== 0) {
+        joinArr[i] = sep;
+      } else {
+        const joinElement = this[i / 2];
+
+        if ((joinElement !== null) && (joinElement !== undefined)) {
+          joinArr[i] = joinElement;
+        } else {
+          joinArr[i] = '';
+        }
+      }
+    }
+
+    for (const i of joinArr) {
+      joinStr += i;
+    }
```

Comment:
simplify your code all what you need is set `','` as default value for `separator` parametr then iterate through string and check on each iteration whether the current character is neither `null ` nor `undefined` if no, concate current char with result string, then check if current index is not last, and concate separator with result string. That's it